src/main/java/com/google/opensesame/projects/QueryFilter.java

```diff
@@ -0,0 +1,88 @@
+package com.google.opensesame.projects;
+
+import com.google.opensesame.util.ServletValidationException;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A Datastore filter stored in the form used by Objectify when loading entities from the Datastore.
+ *
+ * <p>See here for how Objectify conducts filtering on a query:
+ * https://www.javadoc.io/static/com.googlecode.objectify/objectify/6.0.6/com/googlecode/objectify/cmd/Query.html#filter(java.lang.String,java.lang.Object)
+ */
+class QueryFilter {
+  public static final String FILTER_QUERY_REGEX = "^[A-Za-z]+ (>|>=|!=|=|<|<=) .+$";
+
+  /**
+   * Parses a QueryFilter from a string and responds with errors to the provided servlet response
+   * object.
+   *
+   * <p>QueryFilter strings should be in the format "fieldName <comparator> value". Examples:
+   *
+   * <blockquote>
+   *
+   * <pre>
+   * "numMentors >= 2"
+   * "numInterestedUsers = 0"
+   * </pre>
+   *
+   * </blockquote>
+   *
+   * @param QueryFilterString The filter query string.
+   * @param response The servlet response to send errors to.
+   * @return Returns the QueryFilter or null if there was an error parsing the string.
+   * @throws IOException
+   */
+  public static QueryFilter fromString(String QueryFilterString)
+      throws IOException, ServletValidationException {
+    if (!QueryFilterString.matches(FILTER_QUERY_REGEX)) {
+      throw new ServletValidationException(
+          "Invalid filter query.",
+          "Unable to query for projects.",
+          HttpServletResponse.SC_BAD_REQUEST);
+    }
+
+    String[] splitFilterRequest = QueryFilterString.split(" ");
+    String filterFieldName = splitFilterRequest[0];
+    Optional<Field> filterField =
+        ProjectQuery.queryableFields.stream()
+            .filter((field) -> field.getName().equals(filterFieldName))
+            .findFirst();
+    if (!filterField.isPresent()) {
+      throw new ServletValidationException(
+          "Cannot filter by the field '" + filterFieldName + "'.",
+          "Unable to query for projects.",
+          HttpServletResponse.SC_BAD_REQUEST);
+    }
+
+    String comparisonValue = splitFilterRequest[2];
```

Comment:
using `comparison` in this variable name confuses me a lot. It took me a couple minutes to realize that this variable is the comparator. Just call this variable `value` to stay consistent with your docstring that states that `QueryFilterString` will come in the format "fieldName \<comparator\> value".
```suggestion
    String value = splitFilterRequest[2];
```

Reply:
I agree. Sorry it took so long to read! Will change.